gr-blocks/include/gnuradio/blocks/tag_share.h

```diff
@@ -0,0 +1,68 @@
+/* -*- c++ -*- */
+/* 
+ * Copyright 2017 Free Software Foundation, Inc.
+ * 
+ * This file is part of GNU Radio
+ * 
+ * GNU Radio is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3, or (at your option)
+ * any later version.
+ * 
+ * GNU Radio is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with GNU Radio; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+
+#ifndef INCLUDED_BLOCKS_TAG_SHARE_H
+#define INCLUDED_BLOCKS_TAG_SHARE_H
+
+#include <gnuradio/blocks/api.h>
+#include <gnuradio/sync_block.h>
+
+namespace gr {
+  namespace blocks {
+
+    /*!
+     * \brief Adds tags from Input 1 onto Input 0's stream.
+     * \ingroup blocks
+     *
+     * \details
+     * This block utilizes the GNU Radio runtime's tag propagation policy
+     * to transfer or share Input 1's tags to Input 0's stream. This is
+     * useful when a signal is detected via a correlate_access_code_bb or
+     * a threshold crossing from a complex_to_mag_squared block. The tag from
+     * that detection is on the alternate stream, either bytes or floats.
+     * Often there is further signal processing that should be done on the 
+     * complex stream. This block allows the detection tags to be added to
+     * the complex stream to trigger downstream processing without the need of
+     * redundant trigger inputs on all subsequent blocks.
+     */
+    class BLOCKS_API tag_share : virtual public gr::sync_block
+    {
+     public:
+      typedef boost::shared_ptr<tag_share> sptr;
+
+      /*!
+       * \brief Return a shared_ptr to a new instance of blocks::tag_share.
+       *
+       * To avoid accidental use of raw pointers, blocks::tag_share's
+       * constructor is in a private implementation
+       * class. blocks::tag_share::make is the public interface for
+       * creating new instances.
+       */
+      static sptr make(size_t sizeof_stream_item1, size_t sizeof_stream_item2);
```

Comment:
Adding a third parameter, vlen, with a default argument of 1, would allow this block to work more easily on streams of vectors.  One could accomplish that as-is by instantiating the block with a itemsize*vlen argument, but this is prone to error and is less readable.